spec-main/api-context-bridge-spec.ts

```diff
@@ -62,977 +62,2051 @@ describe('contextBridge', () => {
+              await root.example.myPromise;
+              return null;
+            } catch (err) {
+              return err;
+            }
+          });
+          expect(result).to.be.an.instanceOf(Error).with.property('message', 'i-rejected');
+        });
+
+        it('should proxy promises and resolve with the correct value if it resolves later', async () => {
+          await makeBindingWindow(() => {
+            contextBridge.exposeInMainWorld('example', {
+              myPromise: () => new Promise(resolve => setTimeout(() => resolve('delayed'), 20))
+            });
+          });
+          const result = await callWithBindings((root: any) => {
+            return root.example.myPromise();
+          });
+          expect(result).to.equal('delayed');
+        });
+
+        it('should proxy nested promises correctly', async () => {
+          await makeBindingWindow(() => {
+            contextBridge.exposeInMainWorld('example', {
+              myPromise: () => new Promise(resolve => setTimeout(() => resolve(Promise.resolve(123)), 20))
+            });
+          });
+          const result = await callWithBindings((root: any) => {
+            return root.example.myPromise();
+          });
+          expect(result).to.equal(123);
+        });
+
+        it('should proxy methods', async () => {
+          await makeBindingWindow(() => {
+            contextBridge.exposeInMainWorld('example', {
+              getNumber: () => 123,
+              getString: () => 'help',
+              getBoolean: () => false,
+              getPromise: async () => 'promise'
+            });
+          });
+          const result = await callWithBindings(async (root: any) => {
+            return [root.example.getNumber(), root.example.getString(), root.example.getBoolean(), await root.example.getPromise()];
+          });
+          expect(result).to.deep.equal([123, 'help', false, 'promise']);
+        });
+
+        it('should proxy functions', async () => {
+          await makeBindingWindow(() => {
+            contextBridge.exposeInMainWorld('example', () => 'return-value');
+          });
+          const result = await callWithBindings(async (root: any) => {
+            return root.example();
+          });
+          expect(result).equal('return-value');
+        });
+
+        it('should not double-proxy functions when they are returned to their origin side of the bridge', async () => {
+          await makeBindingWindow(() => {
+            contextBridge.exposeInMainWorld('example', (fn: any) => fn);
+          });
+          const result = await callWithBindings(async (root: any) => {
+            const fn = () => null;
+            return root.example(fn) === fn;
+          });
+          expect(result).equal(true);
+        });
+
+        it('should properly handle errors thrown in proxied functions', async () => {
+          await makeBindingWindow(() => {
+            contextBridge.exposeInMainWorld('example', () => {
```

Comment:
the exposeInMainWorld tests still look as if these have been formatted. IIUC, there shouldn't be a diff for the existing tests.

Reply:
I have contextualized a few initializations and added a grouping over existing test cases; that's why there might also be some differences in the current test cases.
I will cross-check the styling once again 🙇‍♂️